core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -398,9 +401,33 @@ class HttpGoogleServicesDAO(
 
   override def getBucket(bucketName: String)(implicit executionContext: ExecutionContext): Future[Option[Bucket]] = {
     implicit val service = GoogleInstrumentedService.Storage
-    val getter = getStorage(getBucketServiceAccountCredential).buckets().get(bucketName)
-    retryWithRecoverWhen500orGoogleError(() => { Option(executeGoogleRequest(getter)) }) {
-      case e: HttpResponseException => None
+    retryWithRecoverWhen500orGoogleError(() => {
+      val getter = getStorage(getBucketServiceAccountCredential).buckets().get(bucketName)
+      Option(executeGoogleRequest(getter))
+    }) {
+      case _: HttpResponseException => None
+    }
+  }
+
+  override def getRegionForRegionalBucket(bucketName: String): Future[Option[String]] = {
+    getBucket(bucketName) map { maybeBucket =>
+      val bucket = maybeBucket.getOrElse(throw new RawlsException(s"Failed to retrieve bucket `$bucketName`"))
+      bucket.getLocationType match {
+        case SingleRegionLocationType => Option(bucket.getLocation.toLowerCase)
+        case _ => None
+      }
+    }
+  }
+
+  override def getComputeZonesForRegion(googleProject: GoogleProjectId, region: String): Future[List[String]] = {
+    implicit val service = GoogleInstrumentedService.Storage
+    retryWithRecoverWhen500orGoogleError(() => {
+      val getter = getComputeManager(getBucketServiceAccountCredential).regions().get(googleProject.value, region)
+      val zonesAsResourceUrls = executeGoogleRequest(getter).getZones.asScala.toList
+      zonesAsResourceUrls.map(_.split("/").last)
+    }) {
+      case e => throw new RawlsException(s"Something went wrong while retrieving zones for region `$region` under Google " +
+        s"project `${googleProject.value}`. Error: ${ExceptionUtils.getMessage(e)}")
```

Comment:
Can you pass the error as a `causedBy` to the `RawlsException` to preserve them both in stack trace hierarchy?

Reply:
I am not sure how to use `causedBy`. So something like this
```
throw new (RawlsException(s"Something went wrong while retrieving zones for region `$region` under Google " +
        s"project `${googleProject.value}`.") causedBy e)
```

Reply:
Most (many?) exceptions accept a second parameter like `new RawlsException(msg: String, causedBy: Throwable)`

Reply:
If `RawlsException` doesn't have that as a constructor argument then maybe we could add it?

Reply:
`RawlsException` does have a `cause` field wherein you can pass the throwable. I have used that 👍